SingularityBase/src/main/java/com/hubspot/singularity/SingularityTaskIdHistory.java

```diff
@@ -15,17 +18,20 @@
   private final Optional<ExtendedTaskState> lastTaskState;
   private final Optional<String> runId;
 
+  private static final Comparator<SingularityTaskHistoryUpdate> comparator = new Comparator<SingularityTaskHistoryUpdate>() {
```

Comment:
We can't have two separate state updates for the same state, so shouldn't need that